src/main/java/de/rytrox/varo/teams/spawnpoint/SpawnPointService.java

```diff
@@ -0,0 +1,92 @@
+import de.rytrox.varo.database.entity.TeamMember;
+import de.rytrox.varo.database.enums.PlayerStatus;
+import de.rytrox.varo.database.repository.TeamMemberRepository;
+import de.rytrox.varo.gamestate.GameStateHandler;
+import de.rytrox.varo.teams.events.TeamMemberDisconnectEvent;
+import de.rytrox.varo.teams.events.TeamMemberSpawnEvent;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+import java.util.logging.Level;
+
+public class SpawnPointService implements Listener {
+
+    private final GameStateHandler gameStateHandler = GameStateHandler.getInstance();
+
+    private final Varo main;
+
+    private final TeamMemberRepository teamMemberRepository;
+
+    public SpawnPointService(@NotNull Varo main, @NotNull TeamMemberRepository teamMemberRepository) {
+        this.main = main;
+        this.teamMemberRepository = teamMemberRepository;
+    }
+
+    public void setSpawnPoint(Player executor, String targetName) {
+        Bukkit.getScheduler().runTaskAsynchronously(main, () -> {
+            // get Target
+            TeamMember member = teamMemberRepository.getPlayer(Bukkit.getOfflinePlayer(targetName));
+
+            if(member != null) {
+                SpawnPoint point = Optional.ofNullable(member.getSpawnPoint())
+                        .orElse(new SpawnPoint());
+
+                point.setLocation(executor.getLocation());
+                member.setSpawnPoint(point);
+
+                // If a player has a spawnpoint, it is registered
+                if(member.getStatus() == PlayerStatus.NOT_REGISTERED)
+                    member.setStatus(PlayerStatus.ALIVE);
+
+                main.getDB().update(member);
+                executor.sendMessage(ChatColor.translateAlternateColorCodes('&',
+                        String.format("&8[&6Varo&8] &7Du hast den Spawnpoint von %s erfolgreich gesetzt!", member.getOfflinePlayer().getName())));
+                main.getLogger().log(Level.INFO, String.format("Player %s sets Spawnpoint of %s to %s", executor.getName(), member.getOfflinePlayer().getName(), point));
+            } else executor.sendMessage(ChatColor.RED + "Dieser Spieler ist nicht bekannt");
+        });
+    }
+
+    /**
+     * Teleport the player when the player joins and the game is running
+     *
+     * @param event the TeamMemberSpawnEvent
+     */
+    @EventHandler
+    public void onSpawn(TeamMemberSpawnEvent event) {
+        Player player = event.getPlayer();
+
+        // Only teleport if the game is running and the player is alive
+        if(event.getMember().getStatus() == PlayerStatus.ALIVE &&
+                (gameStateHandler.getCurrentGameState() == GameStateHandler.GameState.MAIN ||
+                 gameStateHandler.getCurrentGameState() == GameStateHandler.GameState.FINAL)) {
```

Comment:
Das sind hier die falschen Gamestates. 
Ich denke du willst hier die Phase vor bzw. während dem Countdown (PRE_GAME)
und die Siegerehrung (POST was man btw. mal zu POST_GAME umbenennen könnte). 

Im Ticket zu den Gamestates findest du Kommentare dazu, was ich mir bei welchem Gamestate überlegt habe. Hätte das aber sinnvollerweise auch mal im Sourcecode hinterlegen können...

Reply:
Nein tatsächlich hatte ich mir überlegt, dass diese SpawnPoints die aktuellen SpawnPoints sind und nicht nur die initialien. Damit können wir auch beim BorderChange detecten, welche Leute out of Range sind. Daher hab ich diesen EventListener geschrieben, dass der den Spieler während das Spiel läuft, teleportiert.